core/checksums/src/main/java/software/amazon/awssdk/checksums/Checksum.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.checksums;
+
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+
+/**
+ * An enumeration of supported checksums.
+ */
+@SdkProtectedApi
+public enum Checksum {
+    CRC32C,
+    CRC32,
+    MD5,
+    SHA256,
+    SHA1,
+}
```

Comment:
This is going to sound weird, but can we make this a class with public static final fields for the checksum values?

```
public final class Checksum {
    public final class Checksum CRC32C = Checksum.of("CRC32C");
    public final class Checksum CRC32 = Checksum.of("CRC32");
    public final class Checksum MD5 = Checksum.of("MD5");
    public final class Checksum SHA256 = Checksum.of("SHA256");
    public final class Checksum SHA1 = Checksum.of("SHA1");

    private static final Map<String, Checksum> VALUES = new ConcurrentHashMap<>();

    private final Checksum of(String name) {
        return VALUES.computeIfAbsent(name, n -> new Checksum());
    }
}
```

Making it not an enum allows us to add more in the future from within an SDK plugin. For example, S3 can define `Checksum.of("SHA512")` without needing to have it pushed "upstream" to the `Checksum` type. Making that useful will also require additional methods being added to Checksum so that we can interact with it polymorphically, (e.g. to get an implementation of the actual checksum algorithm), but making it not-an-enum now will allow for that.

Reply:
Sounds good to me.